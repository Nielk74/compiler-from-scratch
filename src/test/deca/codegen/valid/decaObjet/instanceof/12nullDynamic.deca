// Description:
//    instanceof sur un objet dont le type dynamique est null
//    instanceof sur l'objet null
//
// Resultats:
//    a is instance of B
//    null is instance of Object
//
// Historique:
//    cree le 21/06/2022

class A {}

class B {}

{
    A a = null;

    if (a instanceof B) {
        println("a is instance of B");
    } else {
        println("a is not instance of B");
    }

    if (null instanceof Object) {
        println("null is instance of Object");
    } else {
        println("null is not instance of Object");
    }
}