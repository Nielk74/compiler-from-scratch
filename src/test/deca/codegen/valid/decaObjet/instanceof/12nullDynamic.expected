a is instance of B
null is instance of Object
